# Generated by Django 3.0.7 on 2021-03-04 13:57
from django.conf import settings
from django.db import migrations
from django.utils import timezone
from requests import exceptions

from restaurateur.coordinates import fetch_coordinates


def create_objects(old_model, new_model):
    for item in old_model.objects.all():
        place, created = new_model.objects.get_or_create(address=item.address)
        try:
            place.lat, place.lon = fetch_coordinates(settings.GEO_API_KEY, item.address)
        except exceptions.HTTPError:
            place.lat, place.lon = None, None
        place.fetch_at = timezone.now()
        place.save()


def add_coordinates(apps, schema_editor):
    Order = apps.get_model('foodcartapp', 'Order')
    Restaurant = apps.get_model('foodcartapp', 'Restaurant')
    Place = apps.get_model('foodcartapp', 'Place')
    create_objects(Order, Place)
    create_objects(Restaurant, Place)


class Migration(migrations.Migration):

    dependencies = [
        ('foodcartapp', '0048_place'),
    ]

    operations = [
        migrations.RunPython(add_coordinates),
    ]
